### ISSUE TITLE
Weird tooltip positioning

### ISSUE THREAD

**Issue by mlaggner** [JFormDesigner/FlatLaf#164]
I've created a window with an error indicator in the lower right corner which shows a tooltip (flat JButton) when hovered with the mouse like 
![Screenshot_20200824_180436](https://user-images.githubusercontent.com/1523778/91068543-be15a980-e634-11ea-9c98-f2e88ffeccf3.png)

but when I maximize the window (or move it to the lower right corner of the screeen), the tooltip hoveres the JButton which become unclickable
![Screenshot_20200824_180506](https://user-images.githubusercontent.com/1523778/91068672-ea312a80-e634-11ea-898c-b8efae24ce81.png)

doing the same in IntelliJ (which is a swing UI too), this behaves as intended:
![Screenshot_20200824_180929](https://user-images.githubusercontent.com/1523778/91068781-0cc34380-e635-11ea-8a90-9ea61caf49e3.png)
![Screenshot_20200824_181020](https://user-images.githubusercontent.com/1523778/91068808-16e54200-e635-11ea-9e56-8d50a19914c3.png)

is there anything we can do by the LaF?

**Comment by DevCharly** [comment]
Yep, that positioning is weird...

Class `javax.swing.ToolTipManager` does the tooltip positioning.
It simply checks whether the tooltip is (partly) outside of the screen and then moves x,y so that it is inside.

See https://github.com/openjdk/jdk/blob/bcff4990ad9db20d11c59841a409717118645a65/src/java.desktop/share/classes/javax/swing/ToolTipManager.java#L321-L334

IntelliJ IDEA does not use `javax.swing.ToolTipManager`.
They use [IdeTooltipManager](https://github.com/JetBrains/intellij-community/blob/master/platform/platform-impl/src/com/intellij/ide/IdeTooltipManager.java) and [HelpTooltipManager](https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/ide/HelpTooltipManager.java) and [HelpTooltip](https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/ide/HelpTooltip.java) with [RelativePoint](https://github.com/JetBrains/intellij-community/blob/master/platform/util/ui/src/com/intellij/ui/awt/RelativePoint.java),
which seems to do the out-of-screen position fix if necessary.

So there is no easy fix for this.

Will try to solve it in [FlatPopupFactory.getPopup()](https://github.com/JFormDesigner/FlatLaf/blob/535c3ddf6c425548c6eb73678358d0e73473ecfc/flatlaf-core/src/main/java/com/formdev/flatlaf/ui/FlatPopupFactory.java#L59), which is used to display tooltips (and other popups). 
But need the x,y location of the mouse pointer (from the last `MouseEvent`), which is not passed to this method...

**Comment by DevCharly** [comment]
@mlaggner in case of your notification button, which is always in lower-right corner of window, it would be better to show the notification tooltip always above the button (as in IntelliJ IDEA). You can do this by overriding `Point getToolTipLocation(MouseEvent event)` of the button and compute the tip location yourself. Create a temporary tooltip to compute the preferred size of the tooltip. E.g.

~~~java
Point getToolTipLocation(MouseEvent event) {
    JToolTip tip = new JToolTip();
    tip.setTipText( getToolTipText() );
    Dimension tipSize = tip.getPreferredSize();

    // TODO compute tip location based on tipSize and upper-right corner of button
}
~~~

**Comment by mlaggner** [comment]
thanks for your great help (once more). 
I managed to create a custom tooltip location for all components in the status bar with `getToolTipLocation` and some code of [HelpTooltip.java](https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/ide/HelpTooltip.java).

Nevertheless, it would be great if the LaF could manage such corner cases!

**Comment by DevCharly** [comment]
fixed in 0.42
